test/integration/query-interface.test.js

```diff
@@ -483,8 +491,10 @@ describe(Support.getTestDialectTeaser('QueryInterface'), () => {
   describe('constraints', () => {
     beforeEach(async function() {
       this.User = this.sequelize.define('users', {
-        username: DataTypes.STRING,
-        email: DataTypes.STRING,
+        // Db2 does not allow unique constraint for a nullable column, Db2
+        // throws SQL0542N error if we create constraint on nullable column.
+        username: dialect === 'db2' ? { type: DataTypes.STRING, allowNull: false } : DataTypes.STRING,
```

Comment:
why?

Reply:
It is restricted by Db2 server.

Reply:
@sdepold [Db2 doc](https://www.ibm.com/docs/en/db2/11.5?topic=constraints-unique) clearly mentions that - The columns specified in a unique constraint must be defined as NOT NULL.